oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -101,31 +103,41 @@ public void setIdToken(String idToken) {
     this.idToken = idToken;
   }
 
+  public MockLowLevelHttpRequest getRequest() {
+    return request;
+  }
+
   @Override
   public LowLevelHttpRequest buildRequest(String method, String url) throws IOException {
     if (url.startsWith(ComputeEngineCredentials.getTokenServerEncodedUrl())) {
-      return getMockRequestForTokenEndpoint(url);
+      this.request = getMockRequestForTokenEndpoint(url);
+      return this.request;
     } else if (isGetServiceAccountsUrl(url)) {
-      return getMockRequestForServiceAccount(url);
+      this.request = getMockRequestForServiceAccount(url);
+      return this.request;
     } else if (isSignRequestUrl(url)) {
-      return getMockRequestForSign(url);
+      this.request = getMockRequestForSign(url);
+      return this.request;
     } else if (isIdentityDocumentUrl(url)) {
-      return getMockRequestForIdentityDocument(url);
+      this.request = getMockRequestForIdentityDocument(url);
+      return this.request;
```

Comment:
why is this being assigned to request before returning?

Reply:
The request is assigned in this mock transport so that it can be exposed with `getRequest()`, so that headers are easily accessible in tests. Similarly to MockTokenServerTransport [here](https://github.com/googleapis/google-auth-library-java/blob/16510064e861868f649b6bc8fdc54b8a39890812/oauth2_http/javatests/com/google/auth/oauth2/MockTokenServerTransport.java#L179-L194).

Reply:
Ok gotcha. I wish we could use like interceptors or something. Something to fix in the future for sure, but I think this is fine for now.